crates/bitcoin-da/src/verifier.rs

```diff
@@ -288,23 +392,115 @@ impl DaVerifier for BitcoinVerifier {
                 epoch_start_time,
                 block_header.time().secs() as u32,
                 block_header.bits(),
+                network_constants.max_target,
             );
             current_target_bits = target_to_bits(&next_target);
         }
 
-        let total_work = U256::from_be_bytes(previous_light_client_proof_output.da_total_work)
+        let total_work = U256::from_be_bytes(latest_da_state.total_work)
             .saturating_add(&work_add)
             .to_be_bytes();
 
-        Ok(UpdatedDaState {
-            hash: block_header.hash(),
-            height: block_header.height(),
+        Ok(LatestDaState {
+            block_hash: block_header.hash().to_byte_array(),
+            block_height: block_header.height(),
             total_work,
             epoch_start_time,
             prev_11_timestamps,
             current_target_bits,
         })
     }
+
+    fn verify_header_chain_regtest(
```

Comment:
One can get rid of some stuff here, such as the work done per block will always be constant and no need for difficulty change, or calculating the target bytes or compact target for that matter.

Reply:
got rid of unnecessary checks